packages/web/src/providers/gnoswap-theme-provider/GnoswapThemeProvider.tsx

```diff
@@ -1,14 +1,26 @@
 import { Global, ThemeProvider } from "@emotion/react";
-import { useMemo } from "react";
+import { useEffect, useMemo } from "react";
 import { getTheme } from "@utils/theme-utils";
 import globalStyle from "@styles/globalStyle";
 import { useAtom } from "jotai";
 import { ThemeState } from "@states/index";
+import { useWindowSize } from "@hooks/common/use-window-size";
 
 const GnoswapThemeProvider: React.FC<React.PropsWithChildren> = ({
   children,
 }) => {
   const [themeKey] = useAtom(ThemeState.themeKey);
+  const { handleResize, findDeviceType } = useWindowSize();
+
+  useEffect(() => {
```

Comment:
My guess is that you'll be registering and unregistering events frequently. 
It would be nice to improve the hooks.
Additionally, if you don't use the window size, you shouldn't save it.

Reply:
``` 
useEffect(() => {
    window.addEventListener("resize", handleWindowSize);
    return () => {
      window.removeEventListener("resize", handleWindowSize);
    };
  }, []);

  function handleWindowSize() {
    handleBreakpoint(window.innerWidth); 
  }
```

Reply:
good idea. The code has been cleaned up and unnecessary state has been removed.